api/v1alpha1/connection_types.go

```diff
@@ -36,6 +36,13 @@ type ClientConnection struct {
 	// +optional
 	// +notImplementedHide
 	SocketBufferLimit *resource.Quantity `json:"socketBufferLimit,omitempty"`
+
+	// MaxAcceptPerSocketEvent provides configuration for the maximum number of connections to accept from the kernel
+	// per socket event. If there are more than MaxAcceptPerSocketEvent connections pending accept, connections over
+	// this threshold will be accepted in later event loop iterations. If no value is provided Envoy will accept
+	// all connections pending accept from the kernel.
+	// +optional
```

Comment:
```suggestion
    // It is recommended to lower this value for better overload management and reduced per-event cost. 
    // Setting it to 1 is a viable option with no noticeable impact on performance.
	// +optional
	//
```
Although we are not taking the call for the user, I think this note is useful in the API docs, wdyt?